hnn_core/drives.py

```diff
@@ -229,35 +224,33 @@ def _drive_cell_event_times(drive_type, drive_conn, dynamics,
     elif len(matches) > 1:
         raise ValueError('Ambiguous external drive: %s' % drive_type)
 
-    # Return values not checked: False if all weights for given drive type
-    # are zero. Designed to be silent so that zeroing input weights
-    # effectively disables each.
-    n_ampa_nmda_weights = (len(drive_conn['ampa'].keys()) +
-                           len(drive_conn['nmda'].keys()))
-    target_syn_weights_zero = True if n_ampa_nmda_weights == 0 else False
-
     event_times = list()
-    if drive_type == 'poisson' and not target_syn_weights_zero:
-        event_times = _create_extpois(
-            t0=dynamics['tstart'],
-            T=dynamics['tstop'],
-            lamtha=dynamics['rate_constant'][drive_conn['target_type']],
-            prng=prng)
-    elif not target_syn_weights_zero and (drive_type == 'evoked' or
-                                          drive_type == 'gaussian'):
+    if drive_type == 'poisson':
+        # Poisson rate constant in Network.add_poisson_drive() can be
+        # target specific or not
+        if target_type == 'any':
+            rate_constant = dynamics['rate_constant']
+        else:
+            rate_constant = dynamics['rate_constant'][target_type]
+        if rate_constant > 0:
```

Comment:
why allow `rate_constant` < 0 ? Just raise an error, no?

Reply:
That would cause running the `default.json` param file with `add_drive_from_param = True` to fail. `default.json` specifies the creation of a null (zero synaptic weight + a zero rate constant) poisson drive, which was previously ignored due to its null synaptic weight. With my current implementation, a drive can still get initialized with null synaptic weight, however, checking that the rate constant is not zero prevents `_create_extpois()` from failing without forcing most of our legacy param files to fail.

Side note, we can still decide to prevent drives with null synaptic weight from getting initialized if we wish, I would just propose that we do so by adding checks with error messages rather than silently ignoring them.

Reply:
An alternative to the 'if' statement above would be to add some logic to `params_default.py` that changes the rate constant to >0 if all the synaptic weights are zero. Then, we could raise an error here if `rate_constant < 0`, per your suggestion. WDYT?

Reply:
In line with my [comment below](https://github.com/jonescompneurolab/hnn-core/pull/383#discussion_r671903614), idiosyncracies in the handling of param files should be separated from the API itself. If you see a zero rate constant and don't want to raise an error, don't create the drive dictionary when parsing the param file?